core-processor/src/main/java/io/micronaut/inject/beans/visitor/EvaluatedExpressionContextTypeElementVisitor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.inject.beans.visitor;
+
+import io.micronaut.context.annotation.EvaluatedExpressionContext;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.expressions.context.ExpressionWithContext;
+import io.micronaut.inject.ast.ClassElement;
+import io.micronaut.inject.visitor.TypeElementVisitor;
+import io.micronaut.inject.visitor.VisitorContext;
+import io.micronaut.inject.writer.ClassGenerationException;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * A {@link TypeElementVisitor} that visits classes annotated with
+ * {@link EvaluatedExpressionContext} and produces
+ * {@link ExpressionWithContext} instances at compilation time.
+ *
+ * @author Sergey Gavrilov
+ * @since 4.0.0
+ */
+@Internal
+public class EvaluatedExpressionContextTypeElementVisitor implements TypeElementVisitor<Object, Object> {
```

Comment:
override `getAnnotationNames()` and add `EvaluatedExpressionContext.class`

Also override `getVisitorKind()` and return ISOLATING

Reply:
Make `final` too